diff --git a/kernel/sched/core.c b/kernel/sched/core.c
index abcdef1..abcdef2 100644
--- a/kernel/sched/core.c
+++ b/kernel/sched/core.c
@@ -8730,11 +8730,11 @@ static void uclamp_set(struct cgroup_subsys_state *css)
	static struct uclamp_param tgts[] = {
-		{"top-app",            "10",  "max",  1, 1, 20480},
-		{"rt",			"0",  "max",  1, 0, 20480},
-		{"nnapi-hal",		"0",  "max",  1, 0, 20480},
-       		{"foreground",          "0",  "max",  1, 0, 20480},
-                {"camera-daemon",      "10",  "max",  1, 1, 20480},
-                {"system",              "0",  "max",  0, 0, 20480},
-                {"dex2oat",             "0",  "60",   0, 0,   512},
-        	{"background",          "0",  "50",   0, 0,  1024},
-        	{"system-background",   "0",  "50",   0, 0,  1024},
+		{"top-app",            "12",  "75",   0, 0, 20480},
+		{"rt",                 "0",   "max",  1, 0, 20480},
+		{"nnapi-hal",          "0",   "80",   0, 0, 20480},
+		{"foreground",         "6",   "60",   0, 0, 20480},
+		{"camera-daemon",      "10",  "max",  1, 1, 20480},
+		{"system",             "0",   "50",   0, 0, 20480},
+		{"dex2oat",            "0",   "50",   0, 0,   512},
+		{"background",         "0",   "40",   0, 0,  1024},
+		{"system-background",  "0",   "40",   0, 0,  1024},
	};
